ci-operator/step-registry/baremetal/lab/pre/dns/baremetal-lab-pre-dns-commands.sh

```diff
@@ -39,10 +62,30 @@ for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
     echo "Error when parsing the Bare Metal Host metadata"
     exit 1
   fi
-  DNS_FORWARD="${DNS_FORWARD}
+  # shellcheck disable=SC2154
+  if [ "${masters}" -eq 1 ]; then
```

Comment:
why are we re-creating these DNS entries here and why only when one master only?

There shouldn't be this need. The clusters have to be reached by the LoadBalancer, not through the hosts IP.

For single stack, we have to consider the cluster disconnected, which means the Load Balancer will not have the public IP but only one interface, reachable via proxy and that will gather an ipv4 and ipv6 in our private network. This goes back to the vips.yaml file handling.

Reply:
In the case of ABI SNO, a LoadBalancer is not required as the same host IP serves as the LoadBalancer. But yes, we need to consider that adding this condition might cause the single-stack cluster as disconnected.

Reply:
> we need to consider that adding this condition might cause the single-stack cluster as disconnected. 

Sorry, I only meant the cluster to have a cluster-wide proxy to reach the internet (so it's not really a disconnected one), and in general, clients in Prow will need a proxy to reach them when ipv6 single-stack and v6-primary dual-stack. 

See https://github.com/openshift/release/pull/53916#discussion_r1700404822 for further info about the reasons the infra requires the LB in front of any/most confs.